### DISCUSSION TITLE
iptables-ipset-proto6-allports :: disable iptables actions, let only ipset actions

### DISCUSSION THREAD

**Discussion by adriansev** [fail2ban/fail2ban#3455] [Q&A] [answered]
Hi! I'm using ipset blocking and i already have my own tables and match-sets in iptables
and i'm trying to make fail2ban to just push the ips in the service ipset _without_ doing any iptables add/flush.
How can i disable the iptables calls for iptables-ipset-proto6-allports?
Or is there any action that do only ipset actions?
Thanks a lot!

**Comment by adriansev** [comment, marked as answer]
well, i should have tried before doing the question :) it was enough to create iptables-common.local and iptables-ipset-proto6-allports.local in action.d, with all actions that did iptables manipulation set to true (the command `true`)

**Comment by sebres** [comment]
> How can i disable the iptables calls for iptables-ipset-proto6-allports?

Well indirectly you can't. Excepting the trick where you would overwrite iptables parameter, for instance in `iptables-ipset.local`:
```ini
[Init]
iptables = echo iptables

[Init?family=inet6]
iptables = echo ip6tables
```

Or simply overwrite all iptables invocations, e. g. parameters `actionstart`, `actionstop`, `actionflush`, etc.

> Or is there any action that do only ipset actions?

I don't know such an action, but it is simply to create own ipset action - basically you'd need `actionban` and `actionunban` only (and may be `actionflush` to unban everything).

```ini
[Definition]
# actionstart = ipset -exist create <ipmset> hash:ip timeout <default-ipsettime> <familyopt>
# actionstop = ipset destroy <ipmset>
actionflush = ipset flush <ipmset>
actionban = ipset -exist add <ipmset> <ip> timeout 0
actionunban = ipset -exist del <ipmset> <ip>

[Init]
ipmset = f2b-<name>
familyopt =

[Init?family=inet6]
ipmset = f2b-<name>6
familyopt = family inet6
```

**Comment by adriansev** [comment]
while i solved my problems myself after figuring out the way the `action` verbs works, i see that you mention `iptables-ipset` which is not present on my fail2ban versions (0.11.2 on centos 7 and 1.0.2 on fedora), so now my question is: given relative simplicity of verbs/actions organization, would be reliable to use any kind of action found in upstream master(here) with whichever version of fail2ban is available on the host? 
Thanks a lot!

**Reply by sebres** [reply]
> would be reliable to use any kind of action found in upstream master(here) with whichever version of fail2ban is available on the host?

Conditionally. Unfortunately, they are not always compatible... Especially by 0.11 to 1.0 update.
For example because of behavior changing of `actioncheck` (see #488).

**Reply by adriansev** [reply]
oh, thanks a lot for info!!
